content/working_with/manager/using-secrets.md

```diff
@@ -6,10 +6,18 @@ draft: false
 weight: 875
 ---
 
-The secrets store provides a variable storage (key-value pairs) for data that you do not want to expose in plain text in Cloudify blueprints, such as login credentials for a platform.
-When you create a secret, the key value can be a text string or it can be a file that contains the key value. With the secrets store, you can  make sure all secrets (for example credentials to IaaS environments) are stored separately from blueprints, and adhere to isolation requirements between different tenants. You can include the secret's key in your blueprints and not include the actual values in the blueprints.
+The secrets store provides a secured variable storage (key-value pairs) for data that you do not want to expose in plain text in Cloudify blueprints, such as login credentials for a platform.
+The secrets values are being stored encrypted in our DB. We use the Fernet encryption algorithm of cryptography library,
+it's a symmetric encryption method which makes sure that the message encrypted cannot be manipulated/read without the key.
+When you create a secret, the key value can be a text string or it can be a file that contains the key value. With the secrets store, you can make sure all secrets (for example credentials to IaaS environments) are stored separately from blueprints, and adhere to isolation requirements between different tenants. You can include the secret's key in your blueprints and not include the actual values in the blueprints.
 For more information, see the [get_secret]({{< relref "developer/blueprints/spec-intrinsic-functions.md#get-secret" >}}) intrinsic function.
 
+### Hidden value secret
+
+All the secrets values are encrypted in the DB, when you create a secret you can specify if you want it’s value to be hidden.
+A hidden-value secret means the secret value is only shown to the user that created the secret and to admins.
+Use of the secret is allowed according to user roles and visibility setting for the secret.
+
```

Comment:
A hidden-value secret means the secret's value will only be presented to the user who created it, tenant managers and sys-admins. Use of the secret is allowed according to the users roles, and visibility of the secret.